crates/hyperqueue/src/dashboard/data/data.rs

```diff
@@ -23,11 +21,23 @@ pub struct DashboardData {
     alloc_timeline: AllocationTimeline,
     /// Determines the active time range
     time_mode: TimeMode,
+    /// Is streaming from a live server enabled?
+    stream_enabled: bool,
 }
 
 impl DashboardData {
+    pub fn new(time_mode: TimeMode, stream_enabled: bool) -> Self {
+        Self {
+            worker_timeline: Default::default(),
+            job_timeline: Default::default(),
+            alloc_timeline: Default::default(),
+            time_mode,
+            stream_enabled,
+        }
+    }
+
     pub fn push_new_events(&mut self, mut events: Vec<Event>) {
-        events.sort_unstable_by_key(|e| e.time());
+        events.sort_unstable_by_key(|e| e.time);
```

Comment:
This sort is probably wrong and not necessary. Server sends the messages in the correct logical order and this sort make break the correct ordering. Because: 
* Time may go in past between server restarts
* It is unstable sort and some events may have same time, naturally or because of ms precision

Reply:
I wasn't sure before if the server really upholds this constraint. I removed the sort, thanks.